controller/src/controller/services/builder/fsadmin.py

```diff
@@ -0,0 +1,149 @@
+"""Construction of Kubernetes objects for user fileservers."""
+
+from __future__ import annotations
+
+from kubernetes_asyncio.client import (
+    V1Container,
+    V1ObjectMeta,
+    V1PersistentVolumeClaim,
+    V1Pod,
+    V1PodSecurityContext,
+    V1PodSpec,
+    V1SecurityContext,
+)
+from structlog.stdlib import BoundLogger
+
+from ...config import (
+    FSAdminConfig,
+    PVCVolumeSource,
+    VolumeConfig,
+)
+from ...constants import ARGO_CD_ANNOTATIONS
+from ...models.domain.fsadmin import (
+    FSAdminObjects,
+)
+from .volumes import VolumeBuilder
+
+__all__ = ["FSAdminBuilder"]
+
+
+class FSAdminBuilder:
+    """Construct Kubernetes objects for user file servers.
+
+    Parameters
+    ----------
+    volumes
+        Volumes to mount in the user's file server.
+    """
+
+    def __init__(
+        self,
+        config: FSAdminConfig,
+        volumes: list[VolumeConfig],
+        logger: BoundLogger,
+    ) -> None:
+        self._config = config
+        self._volumes = volumes
+        self._logger = logger
+        self._volume_builder = VolumeBuilder()
+
+    def build(self) -> FSAdminObjects:
+        """Construct the objects that make up fsadmin.
+
+        Returns
+        -------
+        FSAdminObjects
+            Kubernetes objects for the fsadmin environment.
+        """
+        return FSAdminObjects(
+            pvcs=self._build_pvcs(),
+            pod=self._build_pod(),
+        )
+
+    def _build_metadata(self, name: str, suffix: str = "") -> V1ObjectMeta:
+        """Construct the metadata for an object.
+
+        This adds some standard labels and annotations providing Nublado
+        metadata and telling Argo CD how to handle this object.
+        """
+        labels = {
+            "nublado.lsst.io/category": "fsadmin",
+        }
+        annotations = ARGO_CD_ANNOTATIONS.copy()
+        return V1ObjectMeta(
+            name=name + suffix,
+            labels=labels,
+            annotations=annotations,
+        )
+
+    def _build_pod(self) -> V1Pod:
+        """Construct the pod for fsadmin."""
+        wanted_volumes = {m.volume_name for m in self._config.volume_mounts}
+        volumes = self._volume_builder.build_volumes(
+            (v for v in self._volumes if v.name in wanted_volumes),
+            pvc_prefix=self._config.pod_name,
+        )
+        mounts = self._volume_builder.build_mounts(
+            self._config.volume_mounts,
+            prefix=self._config.mount_prefix
+            if self._config.mount_prefix
+            else "",
```

Comment:
A trinary if statement here is pretty unreadable when it has to be broken across three lines. I would pull this out into a temporary variable to avoid the weird formatting.